### PR TITLE
Correctly set runtime search paths for OSX unit test bundles

### PR DESCRIPTION
**Pull request by dnkoutso**

### CODE REVIEW

File: lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb [hunk 1]
```diff
@@ -217,8 +217,9 @@ def generate_ld_runpath_search_paths
           ld_runpath_search_paths = ['$(inherited)']
           if target.platform.symbolic_name == :osx
             ld_runpath_search_paths << "'@executable_path/../Frameworks'"
+            symbol_type = target.user_targets.map(&:symbol_type).uniq.first
```
Comment (dnkoutso) [inline lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb]: Is it possible that an `aggregate_target` integrate more than one `user_target`? If so what is the scenario I can use to test this out and what should the expected behavior be?

**Comment by dnkoutso** [comment]
Will need to bump integration specs.

**Comment by benasher44** [comment]
My only thought is maybe abstract targets can have multiple user targets? I haven't verified that though.

**Comment by benasher44** [comment]
I think in that case, you'd just skip this for abstract targets, for which I think there is a flag/method you can call to ask that.

**Comment by dnkoutso** [comment]
Will update.

**Comment by dnkoutso** [comment]
@benasher44 I simulated this scenario but `target.user_targets` never returns abstract targets:

```
abstract_target 'Abstract1' do
  abstract_target 'Abstract2' do
    target 'SampleApp Mac Tests' do
      platform :osx, '10.10'
      pod 'OCMock', '~> 3.1.2'
    end
  end 
end
```

**Comment by benasher44** [comment]
I believe target itself would be abstract. The `user_targets` should always be `PBXNativeTarget` types.

**Comment by dnkoutso** [comment]
Ah, looks like still that the `generate_ld_runpath_search_paths` is only invoked for non-abstract targets.

if I print out the `target.name` I see `Pods-Abstract1-Abstract2-SampleApp Mac Tests`

**Comment by DanToml** [comment]
Yep that should all be handled in analyzer/variants. This change LGTM.

**Comment by dnkoutso** [comment]
Btw creating a new target in xcode for a macOS unit test bundle automatically creates this rpath:

screen shot 2017-02-05 at 8 55 04 am

**Comment by benasher44** [comment]
This change looks good to me. Just waiting on merge conflict resolution and green CI

**Comment by dnkoutso** [comment]
@benasher44 the corresponding change for integration specs is here https://github.com/CocoaPods/cocoapods-integration-specs/pull/95

**Comment by DanToml** [comment]
@dnkoutso looks like there is one failure left but I'm not sure why. Maybe needs a rebase?

**Comment by dnkoutso** [comment]
I just did lets see :)

**Comment by dnkoutso** [comment]
Can I merge? :)

**Comment by DanToml** [comment]
Whoop, hadn't seen the ci pass, let ship it

**Pull request merged**